Python package set: major updates

FRidh opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Hydra job

https://hydra.nixos.org/jobset/nixpkgs/python-unstable

FRidh committed:
python: case: 1.5.2 -> 1.5.3

FRidh committed:
python: Cerberus: 0.9.2 -> 1.1

FRidh committed:
python: chainer: 3.4.0 -> 3.5.0

FRidh committed:
python: check-manifest: 0.30 -> 0.36

FRidh committed:
python: CherryPy: 14.0.0 -> 14.0.1

FRidh committed:
python: circus: 0.11.1 -> 0.14.0

FRidh committed:
python: python-cjson: 1.1.0 -> 1.2.1

FRidh committed:
python: clf: 0.5.2 -> 0.5.7

FRidh committed:
python: click: 5.1 -> 6.7

FRidh committed:
python: cloudpickle: 0.2.2 -> 0.5.2

FRidh committed:
python: cmdline: 0.1.6 -> 0.1.8

FRidh committed:
python: cogapp: 2.3 -> 2.5.1

FRidh committed:
python: CoilMQ: 1.0.0 -> 1.0.1

FRidh committed:
python: colour: 0.1.2 -> 0.1.5

FRidh committed:
python: python-consul: 0.7.0 -> 0.7.2

FRidh committed:
python: contexter: 0.1.3 -> 0.1.4

FRidh committed:
python: cookiecutter: 1.4.0 -> 1.6.0

FRidh committed:
python: coveralls: 1.2.0 -> 1.3.0

FRidh committed:
python: cryptography: 2.1.4 -> 2.2.2

FRidh committed:
python: cryptography_vectors: 2.1.4 -> 2.2.2

FRidh committed:
python: cssselect: 0.9.1 -> 1.0.3

FRidh committed:
python: cssutils: 1.0.1 -> 1.0.2

FRidh committed:
python: csvkit: 1.0.2 -> 1.0.3

FRidh committed:
python: cupy: 2.4.0 -> 2.5.0

FRidh committed:
python: cx_Oracle: 6.1 -> 6.2.1

FRidh committed:
python: cytoolz: 0.9.0 -> 0.9.0.1

FRidh committed:
python: dask: 0.17.1 -> 0.17.2

FRidh committed:
python: datadog: 0.10.0 -> 0.20.0

FRidh committed:
python: dbf: 0.96.8 -> 0.97.2

FRidh committed:
python: python-debian: 0.1.23 -> 0.1.32

FRidh committed:
python: deluge-client: 1.2.0 -> 1.3.0

FRidh committed:
python: derpconf: 0.4.9 -> 0.8.2

FRidh committed:
python: detox: 0.10.0 -> 0.11

FRidh committed:
python: python-digitalocean: 1.10.1 -> 1.13.2

FRidh committed:
python: Django: 2.0.2 -> 2.0.4

FRidh committed:
python: django-redis: 4.8.0 -> 4.9.0

FRidh committed:
python: djangorestframework: 3.7.7 -> 3.8.0

FRidh committed:
python: docker-pycreds: 0.2.1 -> 0.2.2

FRidh committed:
python: docker: 2.7.0 -> 3.2.1

FRidh committed:
python: docker-compose: 1.19.0 -> 1.20.1

FRidh committed:
python: dogpile.cache: 0.6.4 -> 0.6.5

FRidh committed:
python: dropbox: 8.0.0 -> 8.7.1

FRidh committed:
python: dulwich: 0.18.6 -> 0.19.0

FRidh committed:
python: ECPy: 0.8.3 -> 0.9.0

FRidh committed:
python: elasticsearch-curator: 5.4.1 -> 5.5.1

FRidh committed:
python: elasticsearch: 6.1.1 -> 6.2.0

FRidh committed:
python: emoji: 0.4.5 -> 0.5.0

FRidh committed:
python: evdev: 0.7.0 -> 0.8.1

FRidh committed:
python: Faker: 0.8.11 -> 0.8.12

FRidh committed:
python: flake8-import-order: 0.17 -> 0.17.1

FRidh committed:
python: fonttools: 3.24.1 -> 3.25.0

FRidh committed:
python: gensim: 3.3.0 -> 3.4.0

FRidh committed:
python: google-api-core: 0.1.4 -> 1.1.0

FRidh committed:
python: google-gax: 0.15.16 -> 0.16.0

FRidh committed:
python: hupper: 1.0 -> 1.1

FRidh committed:
python: interruptingcow: 0.7 -> 0.8

FRidh committed:
python: ipython: 6.2.1 -> 6.3.0

FRidh committed:
python: ipywidgets: 7.1.2 -> 7.2.0

FRidh committed:
python: jdatetime: 1.7.1 -> 2.0.0

FRidh committed:
python: jsonpatch: 1.21 -> 1.23

FRidh committed:
python: jsonrpc-websocket: 0.5 -> 0.6

FRidh committed:
python: Jug: 1.6.4 -> 1.6.5

FRidh committed:
python: jupyter_client: 5.2.2 -> 5.2.3

FRidh committed:
python: kafka-python: 1.4.1 -> 1.4.2

FRidh committed:
python: Keras: 2.1.4 -> 2.1.5

FRidh committed:
python: keyring: 11.0.0 -> 12.0.0

FRidh committed:
python: python-ldap: 2.5.2 -> 3.0.0

FRidh committed:
python: ledgerblue: 0.1.16 -> 0.1.17

FRidh committed:
python: libtmux: 0.7.7 -> 0.8.0

FRidh committed:
python: llfuse: 1.3.2 -> 1.3.3

FRidh committed:
python: lxml: 4.1.1 -> 4.2.1

FRidh committed:
python: m2r: 0.1.13 -> 0.1.14

FRidh committed:
python: magic-wormhole-transit-relay: 0.1.1 -> 0.1.2

FRidh committed:
python: matplotlib: 2.1.2 -> 2.2.2

FRidh committed:
python: moto: 1.2.0 -> 1.3.1

FRidh committed:
python: msgpack-numpy: 0.4.1 -> 0.4.3

FRidh committed:
python: multipledispatch: 0.4.9 -> 0.5.0

FRidh committed:
python: nilearn: 0.4.0 -> 0.4.1

FRidh committed:
python: nipype: 1.0.1 -> 1.0.2

FRidh committed:
python: numpy-stl: 2.3.2 -> 2.4.1

FRidh committed:
python: numpydoc: 0.7.0 -> 0.8.0

FRidh committed:
python: openidc-client: 0.2.0 -> 0.5.0

FRidh committed:
python: openpyxl: 2.5.0 -> 2.5.1

FRidh committed:
python: packaging: 16.8 -> 17.1

FRidh committed:
python: pandas: 0.17.1 -> 0.22.0

FRidh committed:
python: path.py: 11.0 -> 11.0.1

FRidh committed:
python: pbr: 3.1.1 -> 4.0.1

FRidh committed:
python: pendulum: 1.4.2 -> 1.4.4

FRidh committed:
python: phonenumbers: 8.9.0 -> 8.9.2

FRidh committed:
python: Pillow: 5.0.0 -> 5.1.0

FRidh committed:
python: pkginfo: 1.3.2 -> 1.4.2

FRidh committed:
python: plaster_pastedeploy: 0.4.2 -> 0.5

FRidh committed:
python: plotly: 2.4.1 -> 2.5.1

FRidh committed:
python: polib: 1.0.4 -> 1.1.0

FRidh committed:
python: progressbar: 2.2 -> 2.3

FRidh committed:
python: py: 1.5.2 -> 1.5.3

FRidh committed:
python: pybase64: 0.2.1 -> 0.3.1

FRidh committed:
python: pyblake2: 1.1.0 -> 1.1.1

FRidh committed:
python: PyChromecast: 2.0.0 -> 2.1.0

FRidh committed:
python: PyContracts: 1.7.9 -> 1.8.3

FRidh committed:
python: pycparser: 2.14 -> 2.18

FRidh committed:
python: pycryptodome: 3.4.9 -> 3.5.1

FRidh committed:
python: pycryptodomex: 3.4.9 -> 3.5.1

FRidh committed:
python: pyechonest: 8.0.2 -> 9.0.0

FRidh committed:
python: pyfakefs: 3.3 -> 3.4.1

FRidh committed:
python: pylint: 1.8.2 -> 1.8.3

FRidh committed:
python: pyobjc: 4.1 -> 4.2.1

FRidh committed:
python: pyodbc: 4.0.22 -> 4.0.23

FRidh committed:
python: pyroute2: 0.4.21 -> 0.5.0

FRidh committed:
python: pytest-cov: 2.4.0 -> 2.5.1

FRidh committed:
python: pytest-fixture-config: 1.0.1 -> 1.3.0

FRidh committed:
python: pytest-flakes: 1.0.1 -> 2.0.0

FRidh committed:
python: pytest-raisesregexp: 2.0 -> 2.1

FRidh committed:
python: pytest-server-fixtures: 1.1.0 -> 1.3.0

FRidh committed:
python: pytest-shutil: 1.2.8 -> 1.3.0

FRidh committed:
python: pytest-virtualenv: 1.2.7 -> 1.3.0

FRidh committed:
python: pytest: 3.4.2 -> 3.5.0

FRidh committed:
python: pytest-runner: 4.0 -> 4.2

FRidh committed:
python: python-slugify: 1.2.4 -> 1.2.5

FRidh committed:
python: pytools: 2018.1 -> 2018.3

FRidh committed:
python: QtPy: 1.3.1 -> 1.4.0

FRidh committed:
python: readme_renderer: 17.4 -> 18.1

FRidh committed:
python: responses: 0.4.0 -> 0.9.0

FRidh committed:
python: s3fs: 0.0.8 -> 0.1.4

FRidh committed:
python: scipy: 1.0.0 -> 1.0.1

FRidh committed:
python: setuptools_scm: 1.15.6 -> 1.17.0

FRidh committed:
python: Sphinx: 1.7.1 -> 1.7.2

FRidh committed:
python: SQLAlchemy: 1.2.4 -> 1.2.6

FRidh committed:
python: sqlmap: 1.2.3 -> 1.2.4

FRidh committed:
python: tblib: 1.3.0 -> 1.3.2

FRidh committed:
python: testfixtures: 5.4.0 -> 6.0.0

FRidh committed:
python: tiros: 1.0.42 -> 1.0.44

FRidh committed:
python: tornado: 4.5.3 -> 5.0.1

FRidh committed:
python: tox: 2.9.1 -> 3.0.0

FRidh committed:
python: tqdm: 4.19.6 -> 4.20.0

FRidh committed:
python: transaction: 2.1.2 -> 2.2.1

FRidh committed:
python: trezor: 0.9.0 -> 0.9.1

FRidh committed:
python: txaio: 2.8.2 -> 2.9.0

FRidh committed:
python: u-msgpack-python: 2.4.1 -> 2.5.0

FRidh committed:
python: virtualenv: 15.1.0 -> 15.2.0

FRidh committed:
python: vowpalwabbit: 8.4.0 -> 8.5.0

FRidh committed:
python: wakeonlan: 1.0.0 -> 1.1.6

FRidh committed:
python: wheel: 0.30.0 -> 0.31.0

FRidh committed:
python: widgetsnbextension: 3.1.4 -> 3.2.0

FRidh committed:
python: ws4py: 0.4.3 -> 0.5.1

FRidh committed:
python: xarray: 0.10.1 -> 0.10.2

FRidh committed:
python: xcffib: 0.5.1 -> 0.6.0

FRidh committed:
python: yapf: 0.20.2 -> 0.21.0

FRidh committed:
python: zict: 0.1.1 -> 0.1.3

FRidh committed:
python: ZODB: 5.3.0 -> 5.4.0

FRidh committed:
python: zxcvbn-python: 4.4.16 -> 4.4.22

FRidh committed:
python: buildbot-pkg: 1.0.0 -> 1.1.0

FRidh committed:
python: pantsbuild.pants: 1.4.0 -> 1.5.0

FRidh committed:
python: mypy: 0.570 -> 0.580

FRidh committed:
python: pipenv: 10.1.2 -> 11.9.0

FRidh committed:
python: vcstool: 0.1.33 -> 0.1.35

FRidh committed:
python: yq: 2.4.1 -> 2.5.0

FRidh committed:
python: home-assistant-frontend: 20180310.0 -> 20180404.0

FRidh committed:
python: aws-shell: 0.1.1 -> 0.2.0

FRidh committed:
python.pkgs.more-itertools: move expression

FRidh committed:
python.pkgs.backcall: init at 0.1.0

FRidh committed:
python.pkgs.ipython_5: 5.5.0 -> 5.6.0

FRidh committed:
python.pkgs.kiwisolver: init at 1.0.1

FRidh committed:
python.pkgs.pyzmq: move expression

FRidh committed:
python: pyzmq: 16.0.2 -> 17.0.0

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=69412ce9-a639-4f89-b395-cc5a85ca52e4)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ivdcnkynqa01bq6n1srwi1kpjnawifkd-python-2.7.14
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=6a44e8a5-1a8f-482c-9424-cce1456b2e63)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ycr22pph2f6dyvczcyh8v0x0bd7zqgjb-python-2.7.14
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=20e2a911-d48f-409a-86b4-c6f4f23af932)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
copying path '/nix/store/1dxvnqcqxxr5s7iaq4bfc4zjdw2lxzpx-Security-osx-10.9.5' from 'https://cache.nixos.org'...
copying path '/nix/store/dgiipdlscvbw6x44rdmv7v4mzjyi0lmk-bzip2-1.0.6.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/ardl1xch4jxxhqgjvnlmsr0q30ln93hg-gdbm-1.14' from 'https://cache.nixos.org'...
copying path '/nix/store/j0xkvycj96q5gx2ab6zg7156lyjdgqh0-ncurses-6.1' from 'https://cache.nixos.org'...
copying path '/nix/store/6ycjypzsw5ka9r85581g0ryjk7cqql77-openssl-1.0.2o' from 'https://cache.nixos.org'...
copying path '/nix/store/l7d2lzy4sma5hgp1kld6546w2ph6dn64-sqlite-3.22.0' from 'https://cache.nixos.org'...
copying path '/nix/store/frp01ijf344mpfczaajszh8ab2z7mhhb-configd-osx-10.8.5' from 'https://cache.nixos.org'...
copying path '/nix/store/38in1ic2j95a4rxp1szn0p6j06z9z47h-readline-6.3p08' from 'https://cache.nixos.org'...
copying path '/nix/store/1x069g63p15s0mi8av2d0zqpj6hgdssx-python-2.7.14' from 'https://cache.nixos.org'...
/nix/store/1x069g63p15s0mi8av2d0zqpj6hgdssx-python-2.7.14
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=8dbf393c-3ae7-406c-a605-6371ec72554f)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ycr22pph2f6dyvczcyh8v0x0bd7zqgjb-python-2.7.14
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=42a462e5-7fb8-4883-b71d-7c11a92f00f1)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ivdcnkynqa01bq6n1srwi1kpjnawifkd-python-2.7.14
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=c27804d7-4e80-43f3-8809-33080061f550)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
copying path '/nix/store/zhb2xrncr0yd9b5a0x4dbkw953r0xss9-gmp-6.1.2' from 'https://cache.nixos.org'...
copying path '/nix/store/syz6vx1pp0bn2922ngmvny5zzxqbb844-coreutils-8.29' from 'https://cache.nixos.org'...
copying path '/nix/store/xhdi8y85fr535yl1w93979wabdmcgjqs-CF-osx-10.10.5' from 'https://cache.nixos.org'...
copying path '/nix/store/frp01ijf344mpfczaajszh8ab2z7mhhb-configd-osx-10.8.5' from 'https://cache.nixos.org'...
copying path '/nix/store/ardl1xch4jxxhqgjvnlmsr0q30ln93hg-gdbm-1.14' from 'https://cache.nixos.org'...
copying path '/nix/store/6ycjypzsw5ka9r85581g0ryjk7cqql77-openssl-1.0.2o' from 'https://cache.nixos.org'...
copying path '/nix/store/38in1ic2j95a4rxp1szn0p6j06z9z47h-readline-6.3p08' from 'https://cache.nixos.org'...
copying path '/nix/store/l7d2lzy4sma5hgp1kld6546w2ph6dn64-sqlite-3.22.0' from 'https://cache.nixos.org'...
copying path '/nix/store/1x069g63p15s0mi8av2d0zqpj6hgdssx-python-2.7.14' from 'https://cache.nixos.org'...
/nix/store/1x069g63p15s0mi8av2d0zqpj6hgdssx-python-2.7.14
```
</p></details>

Commit:
python.pkgs.Babel: move expression

Commit:
python: Babel: 2.3.4 -> 2.5.3

Commit:
python.pkgs.uritemplate: move expression

Commit:
python: uritemplate: 0.6 -> 3.0.0

Commit:
python.pkgs.uritemplate: update expression

* run tests
* update meta info

Commit:
beancount: move to pythonPackages

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=85cb6c67-0bf1-41f0-bd88-abe64d5142ab)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/1x069g63p15s0mi8av2d0zqpj6hgdssx-python-2.7.14
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=50d903bc-d0f4-41b8-9b0f-dd394dfb5b48)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ivdcnkynqa01bq6n1srwi1kpjnawifkd-python-2.7.14
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=dc28db31-a75b-4319-a37a-f02b5bc82edf)

Attempted: python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ycr22pph2f6dyvczcyh8v0x0bd7zqgjb-python-2.7.14
```
</p></details>

Commit:
python.pkgs.google_api_python_client: move expression

Commit:
python: google-api-python-client: 1.5.1 -> 1.6.6

Commit:
python.pkgs.oauth2client: move expression

Commit:
python: oauth2client: 1.4.12 -> 4.1.2

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=7809ac6d-c676-46ce-b570-3d9f893396b5)

Attempted: beancount, python

<details><summary>Partial log (click to expand)</summary><p>

```
wrapping `/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0/bin/bean-report'...
wrapping `/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0/bin/bean-price'...
wrapping `/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0/bin/bean-identify'...
wrapping `/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0/bin/treeify'...
wrapping `/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0/bin/bean-doctor'...
wrapping `/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0/bin/bean-example'...
wrapping `/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0/bin/bean-web'...
wrapping `/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0/bin/bean-file'...
/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0
/nix/store/1x069g63p15s0mi8av2d0zqpj6hgdssx-python-2.7.14
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=b8d5f739-3098-479e-9726-d94bc661f673)

Attempted: beancount, python

<details><summary>Partial log (click to expand)</summary><p>

```
wrapping `/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0/bin/bean-identify'...
wrapping `/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0/bin/bean-price'...
wrapping `/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0/bin/bean-query'...
wrapping `/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0/bin/bean-report'...
wrapping `/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0/bin/bean-sql'...
wrapping `/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0/bin/bean-web'...
wrapping `/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0/bin/treeify'...
wrapping `/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0/bin/upload-to-sheets'...
/nix/store/lasa5s778y8l1x2hg4yz2npvx6js31r7-python3.6-beancount-2.0.0
/nix/store/ycr22pph2f6dyvczcyh8v0x0bd7zqgjb-python-2.7.14
```
</p></details>

dotlambda commented:
Should I push a libgit2 update to this PR or should we move the pygit2 update to a seperate PR?

dotlambda commented:
Pytest-flake8 is broken due to pytest bump. See https://github.com/tholo/pytest-flake8/issues/45.

Commit:
home-assistant: pin async-timeout version

dotlambda commented:
@FRidh Did you intentionally update python packages that are not located in `pkgs/development/python-modules` such as fava or home-assistant-frontend?

FRidh commented:
> Should I push a libgit2 update to this PR or should we move the pygit2 update to a seperate PR?

We could, or keep it for the next batch. I think it would be nice to stabilize the listed changes, but feel free to add anyway.

> Pytest-flake8 is broken due to pytest bump. See tholo/pytest-flake8#45.

That package always breaks on every update.

> Did you intentionally update python packages that are not located in pkgs/development/python-modules such as fava or home-assistant-frontend?

I did. We should also keep those updated, though maybe some should be excluded. Typically I go over the list of commits and remove commits I don't actually want. I suppose the home-assistant ones should be excluded?

dotlambda commented:
> I think it would be nice to stabilize the listed changes

I don't get what you mean.
The updated `pygit2` version requires an updated `libgit2`. Therefore I think the easiest thing to do is putting the `pygit2` update into a seperate PR.

> I suppose the home-assistant ones should be excluded?

Not necessarily. I usually update the frontend when I updae Home Assistant, but it doesn't hurt if it is updated seperately.

Commit:
python: aiohttp: 3.1.1 -> 3.1.2

Commit:
bonfire: fix build

Loosen version constraints to not have an upper bound.
The upper bounds would have to be updated again on the next keyring
update, so they don't make sense.

Commit:
buku: fix build

Loosen pytest version constraint by applying
https://github.com/jarun/Buku/commit/495d6eac4d9371e8ce6d3f601e2bb9e5e74962b4.

Commit:
python.pkgs.whichcraft: move expression

Commit:
python: whichcraft: 0.1.1 -> 0.4.1

Commit:
python.pkgs.whichcraft: run tests

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=b37aaeb2-378b-4f32-b310-843717186660)

Attempted: beancount, bonfire, buku, home-assistant, python

<details><summary>Partial log (click to expand)</summary><p>

```
tests/components/test_init.py ................                           [ 72%]
tests/components/test_introduction.py .                                  [ 73%]
tests/components/test_logger.py ....                                     [ 75%]
tests/components/test_script.py .....                                    [ 78%]
tests/components/test_shell_command.py ......                            [ 82%]
tests/components/test_system_log.py ...............                      [ 91%]
tests/components/test_websocket_api.py ..............                    [100%]

========================== 163 passed in 4.61 seconds ==========================
[31;1merror:[0m build of '/nix/store/38m2zca50rkjim776rdwjpxxsdg7nrj3-bonfire-unstable-2017-01-19.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=3796ff62-abea-4e93-8dbc-d5379d02e2ec)

Attempted: beancount, home-assistant, python

The following builds were skipped because they don't evaluate on x86_64-darwin: bonfire, buku

<details><summary>Partial log (click to expand)</summary><p>

```
tests/components/test_logger.py ....                                     [ 75%]
tests/components/test_script.py .....                                    [ 78%]
tests/components/test_shell_command.py ......                            [ 82%]
tests/components/test_system_log.py ...............                      [ 91%]
tests/components/test_websocket_api.py ..............                    [100%]

========================== 163 passed in 9.40 seconds ==========================
/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0
/nix/store/gnh4386ljgxgqm1ygcfk5dwrgdb0b0b4-homeassistant-0.65.5
/nix/store/1x069g63p15s0mi8av2d0zqpj6hgdssx-python-2.7.14
```
</p></details>

Commit:
mitmproxy: 3.0.2 -> 3.0.4

Commit:
python.pkgs.pyperclip: move expression

Commit:
python: pyperclip: 1.5.27 -> 1.6.0

Commit:
python.pkgs.wakeonlan: fix build

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=90c667a1-bebd-4fc5-b1ba-30f2d265fe11)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, python

<details><summary>Partial log (click to expand)</summary><p>

```
    0 --:--:-- --:--:--
 --:--:-
-
0

 51 16668   51  8630    0     0   8630      0  0:00:01 --:--:--  0:00:01 13742

100 16668  100 16668    0     0  16668      0  0:00:01 --:--:--  0:00:01 26457

cannot build derivation ‘/nix/store/yqxk9dgzjw2377fq0cq6v8zahy4120mv-python3.6-keyring-12.0.1.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/1cz1casdbbfddyndg8qa4z419rdc15a5-bonfire-unstable-2017-01-19.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/1cz1casdbbfddyndg8qa4z419rdc15a5-bonfire-unstable-2017-01-19.drv’ failed
```
</p></details>

Commit:
python.pkgs.vobject: move expression

Commit:
python.pkgs.txaio: fix build

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=c33f7f27-837e-40c9-b50b-eaefc78c3d41)

Attempted: beancount, home-assistant, mitmproxy, python

The following builds were skipped because they don't evaluate on x86_64-darwin: bonfire, buku

<details><summary>Partial log (click to expand)</summary><p>

```
test/pathod/language/test_http2.py ........................[36m                    [0m
test/pathod/language/test_websockets.py ..........[36m                             [0m
test/pathod/language/test_writer.py ......[36m                                     [0m
test/pathod/protocols/test_http2.py ...........................

[32m[1m================= 1337 passed, 15 deselected in 64.47 seconds ==================[0m
/nix/store/927lil17z0h12lvbziw4dsrin3rllxna-python3.6-beancount-2.0.0
/nix/store/gnh4386ljgxgqm1ygcfk5dwrgdb0b0b4-homeassistant-0.65.5
/nix/store/6n9zlh1rdhsmmh4jv5zvpnl5qir8d451-python3.6-mitmproxy-3.0.4
/nix/store/1x069g63p15s0mi8av2d0zqpj6hgdssx-python-2.7.14
```
</p></details>

Commit:
python.pkgs.dulwich: 0.19.0 -> 0.19.1, fix build

Commit:
python.pkgs.msgpack-numpy: replace msgpack-python by msgpack

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=85571acc-9df5-4772-a238-c25baab16a62)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, python

<details><summary>Partial log (click to expand)</summary><p>

```
tests/components/test_init.py ................                           [ 72%]
tests/components/test_introduction.py .                                  [ 73%]
tests/components/test_logger.py ....                                     [ 75%]
tests/components/test_script.py .....                                    [ 78%]
tests/components/test_shell_command.py ......                            [ 82%]
tests/components/test_system_log.py ...............                      [ 91%]
tests/components/test_websocket_api.py ..............                    [100%]

========================= 163 passed in 28.28 seconds ==========================
[31;1merror:[0m build of '/nix/store/7bq6jngcvwhj4v61x7nks3s3j035khji-bonfire-unstable-2017-01-19.drv' failed
```
</p></details>

Commit:
mkdocs: fix build

Commit:
octoprint: fix build

Commit:
python.pkgs.pyfakefs: run tests

Commit:
pubs: pin pyfakefs version

Commit:
python: sybil: 1.0.7 -> 1.0.8

Commit:
python.pkgs.cerberus: specify checkInputs

Commit:
python.pkgs.datadog: fix build

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=5893b008-8679-4b27-a5b3-b19e535a6fb7)

Attempted: beancount, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

The following builds were skipped because they don't evaluate on x86_64-darwin: bonfire, buku

<details><summary>Partial log (click to expand)</summary><p>

```
tests.test_script.TestScript.test_parse_templates
Test the --parse-templates option. ... ok

----------------------------------------------------------------------
Ran 29 tests in 0.268s

OK
building of '/nix/store/xl7sf65zyyrwmn55jsamxkvi6f35mmfk-python2.7-pyserial-3.4.drv' timed out after 1800 seconds
cannot build derivation '/nix/store/izq3r5awacij6pi39rwkdp79wla2c6h8-OctoPrint-1.3.6.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/izq3r5awacij6pi39rwkdp79wla2c6h8-OctoPrint-1.3.6.drv' failed
```
</p></details>

Commit:
python.pkgs.futures: move expression

Commit:
python: futures: 3.1.1 -> 3.2.0

Commit:
python.pkgs.futures: fix tests

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=63771453-0dc2-4231-a156-42d325680ae9)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```
/build/source/src/octoprint/server/__init__.py:13: ExtDeprecationWarning: Importing flask.ext.babel is deprecated, use flask_babel instead.
  from flask.ext.babel import Babel, gettext, ngettext
/build/source/src/octoprint/server/__init__.py:14: ExtDeprecationWarning: Importing flask.ext.assets is deprecated, use flask_assets instead.
  from flask.ext.assets import Environment, Bundle
...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................SSSSSS....................................................................................
----------------------------------------------------------------------
Ran 633 tests in 77.223s

OK (SKIP=6)
[31;1merror:[0m build of '/nix/store/7bq6jngcvwhj4v61x7nks3s3j035khji-bonfire-unstable-2017-01-19.drv' failed
```
</p></details>

Commit:
python.pkgs.google_api_core: fix build

Commit:
python.pkgs.futures_2_2: uninit

There should only be a single version inside pythonPackages.
This one isn't used anyway.

Commit:
python.pkgs.ldap: fix build

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=48049baf-3bc9-41c8-86aa-3f8caf5723ae)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```
Requirement already satisfied: future in /nix/store/ahz20n8wzx6m8v3mfsn1nqnp5i4cw826-python2.7-future-0.16.0/lib/python2.7/site-packages (from OctoPrint==1.3.6)
Requirement already satisfied: websocket-client in /nix/store/izx1aqyci75sfgkz1qvqbxgl6772m9p3-python2.7-websocket_client-0.47.0/lib/python2.7/site-packages (from OctoPrint==1.3.6)
Requirement already satisfied: requests in /nix/store/ai4yd3bkaffifpcmhzmjwsvm7vh5bkyl-python2.7-requests-2.18.4/lib/python2.7/site-packages (from OctoPrint==1.3.6)
Requirement already satisfied: python-dateutil in /nix/store/kkxdsbad4wvbysjx0pkhc91g3sid6a9f-python2.7-python-dateutil-2.7.2/lib/python2.7/site-packages (from OctoPrint==1.3.6)
Requirement already satisfied: monotonic<1.4,>=1.3 in /nix/store/8jyh42p14dcfc4xzhf9fp6afg7pvy97y-python2.7-monotonic-1.3/lib/python2.7/site-packages (from OctoPrint==1.3.6)
Collecting futures<3.2,>=3.1.1 (from OctoPrint==1.3.6)
  Could not find a version that satisfies the requirement futures<3.2,>=3.1.1 (from OctoPrint==1.3.6) (from versions: )
No matching distribution found for futures<3.2,>=3.1.1 (from OctoPrint==1.3.6)
builder for '/nix/store/a9h0bqx0xmx1ww2348q9wjb32graykk6-OctoPrint-1.3.6.drv' failed with exit code 1
[31;1merror:[0m build of '/nix/store/38m2zca50rkjim776rdwjpxxsdg7nrj3-bonfire-unstable-2017-01-19.drv', '/nix/store/a9h0bqx0xmx1ww2348q9wjb32graykk6-OctoPrint-1.3.6.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=5178eefe-accc-418e-87ae-0ae06e78fad1)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/gpy3r9ss5ngfkib8ylx7jzgahq7m0x5b-patch-shebangs.sh: line 22: warning: command substitution: ignored null byte in input
checking for references to /build in /nix/store/wibnhaqy25y37zi3jxzkp33i98r4dd5h-mkdocs-0.17.3...
wrapping `/nix/store/wibnhaqy25y37zi3jxzkp33i98r4dd5h-mkdocs-0.17.3/bin/mkdocs'...
running install tests
...................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 227 tests in 4.568s

OK
[31;1merror:[0m build of '/nix/store/1fh0a0bk67yw4l4icmc05p55ycwxf657-OctoPrint-1.3.6.drv', '/nix/store/7bq6jngcvwhj4v61x7nks3s3j035khji-bonfire-unstable-2017-01-19.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=7e4a4653-f310-42c3-87f1-a9fc947f091e)

Attempted: beancount, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

The following builds were skipped because they don't evaluate on x86_64-darwin: bonfire, buku

<details><summary>Partial log (click to expand)</summary><p>

```
wrapping `/nix/store/fm84mxgxlbhjpkq5sxqmlnlqv9w9yvdp-mkdocs-0.17.3/bin/mkdocs'...
running install tests
...................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 227 tests in 2.864s

OK
building of '/nix/store/xl7sf65zyyrwmn55jsamxkvi6f35mmfk-python2.7-pyserial-3.4.drv' timed out after 1800 seconds
cannot build derivation '/nix/store/fr5n0mq6wwxpjz0nydxjmapdinrv7arj-OctoPrint-1.3.6.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/fr5n0mq6wwxpjz0nydxjmapdinrv7arj-OctoPrint-1.3.6.drv' failed
```
</p></details>

Commit:
octoprint: fix build

dotlambda commented on pkgs/development/python-modules/ftfy/default.nix:
```diff
@@ -11,7 +11,7 @@
+  version = "5.3.0";
   # ftfy v5 only supports python3. Since at the moment the only
   # packages that use ftfy are spacy and textacy which both support
   # python 2 and 3, they have pinned ftfy to the v4 branch.
```
@FRidh What are your thoughts about this?

FRidh replied:
Ah, right. This commit needs to be removed. I've accidentally merged this one before.

FRidh replied:
Done. I've force-pushed to the branch.

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=50545188-7d18-4470-8fbb-4713a1b3dd35)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```

 generated xml file: /tmp/nix-build-python3.6-pytest-flake8-1.0.0.drv-0/pytest-of-nixbld/pytest-0/test_junit_classname0/TEST.xml 
=========================== 1 passed in 0.12 seconds ===========================

================ 1 failed, 13 passed, 1 xfailed in 2.31 seconds ================

builder for ‘/nix/store/pbm9hvl116kla8pl4r8rkhrzn2nm34xs-python3.6-pytest-flake8-1.0.0.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/yqxk9dgzjw2377fq0cq6v8zahy4120mv-python3.6-keyring-12.0.1.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/1cz1casdbbfddyndg8qa4z419rdc15a5-bonfire-unstable-2017-01-19.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/1cz1casdbbfddyndg8qa4z419rdc15a5-bonfire-unstable-2017-01-19.drv’, ‘/nix/store/rhk51w6zs8g5vw89w5hzl0lrmvjfi1a5-OctoPrint-1.3.6.drv’ failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=932c3e03-9283-4ea8-9167-2ec18b3bc155)

Attempted: beancount, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

The following builds were skipped because they don't evaluate on x86_64-darwin: bonfire, buku

<details><summary>Partial log (click to expand)</summary><p>

```
wrapping `/nix/store/fm84mxgxlbhjpkq5sxqmlnlqv9w9yvdp-mkdocs-0.17.3/bin/mkdocs'...
running install tests
...................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 227 tests in 0.832s

OK
building of '/nix/store/xl7sf65zyyrwmn55jsamxkvi6f35mmfk-python2.7-pyserial-3.4.drv' timed out after 1800 seconds
cannot build derivation '/nix/store/fr5n0mq6wwxpjz0nydxjmapdinrv7arj-OctoPrint-1.3.6.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/fr5n0mq6wwxpjz0nydxjmapdinrv7arj-OctoPrint-1.3.6.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=1358d10c-9ffd-4b26-bd82-4c9694b75bda)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```
test/pathod/language/test_base.py ........................[36m                     [0m
test/pathod/language/test_generators.py ...[36m                                    [0m
test/pathod/language/test_http.py .................................[36m            [0m
test/pathod/language/test_http2.py ........................[36m                    [0m
test/pathod/language/test_websockets.py ..........[36m                             [0m
test/pathod/language/test_writer.py ......[36m                                     [0m
test/pathod/protocols/test_http2.py ...........................

[32m[1m================= 1337 passed, 15 deselected in 208.46 seconds =================[0m
[31;1merror:[0m build of '/nix/store/1fh0a0bk67yw4l4icmc05p55ycwxf657-OctoPrint-1.3.6.drv', '/nix/store/7bq6jngcvwhj4v61x7nks3s3j035khji-bonfire-unstable-2017-01-19.drv', '/nix/store/x6zdylkr9wrr9cxgaq0apdww5s70fa6v-homeassistant-0.65.5.drv' failed
```
</p></details>

Commit:
python.pkgs.moto: fix build

Commit:
python.pkgs.prawcore: 0.12.0 -> 0.15.0

Commit:
python.pkgs.pytest-isort: init at 0.1.0

Commit:
python.pkgs.tinycss2: init at 0.6.1

Commit:
python.pkgs.cssselect2: init at 0.2.1

Commit:
python.pkgs.cairosvg: 2.1.3 -> 2.1.2

The tarball for 2.1.3 is strangely called CairoSVG-2.1.3.linux-x86_64.tar.gz.
Using version 2.1.2 for now.
See https://github.com/Kozea/CairoSVG/issues/188.

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=46390746-2166-4599-8a62-6fa68f29aa28)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```

 generated xml file: /tmp/nix-build-python3.6-pytest-flake8-1.0.0.drv-0/pytest-of-nixbld/pytest-0/test_junit_classname0/TEST.xml 
=========================== 1 passed in 0.12 seconds ===========================

================ 1 failed, 13 passed, 1 xfailed in 2.29 seconds ================

builder for ‘/nix/store/pbm9hvl116kla8pl4r8rkhrzn2nm34xs-python3.6-pytest-flake8-1.0.0.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/yqxk9dgzjw2377fq0cq6v8zahy4120mv-python3.6-keyring-12.0.1.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/1cz1casdbbfddyndg8qa4z419rdc15a5-bonfire-unstable-2017-01-19.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/1cz1casdbbfddyndg8qa4z419rdc15a5-bonfire-unstable-2017-01-19.drv’ failed
```
</p></details>

Commit:
python: pytest-mock: 1.7.1 -> 1.8.0

dotlambda commented:
Libgit2-glib has not released version 0.27 yet and doesn't compile with lingit 0.27 (cf. https://bugzilla.gnome.org/show_bug.cgi?id=794890).
@FRidh Will you remove the pygit2 bump or should I do that?

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=008865db-49c4-4361-b66f-801144b0bd53)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```
-


9
0
3
k

100 2386k  100 2386k    0     0  1193k      0  0:00:02  0:00:02 --:--:-- 1129k

error: build of ‘/nix/store/1cz1casdbbfddyndg8qa4z419rdc15a5-bonfire-unstable-2017-01-19.drv’ failed
```
</p></details>

Commit:
python.pkgs.pytest-fixture-config: fix build

Commit:
python.pkgs.pytest-shutil: add termcolor dependency

Commit:
python.pkgs.pytest-server-fixtures: add future dependency

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=f4de79cb-ff11-4eff-aca2-289c0d3dc930)

Attempted: beancount, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

The following builds were skipped because they don't evaluate on x86_64-darwin: bonfire, buku

<details><summary>Partial log (click to expand)</summary><p>

```
Regression test: Passing the environment to the ManageAssets command ... ok
tests.test_script.TestScript.test_parse_templates
Test the --parse-templates option. ... ok

----------------------------------------------------------------------
Ran 29 tests in 0.826s

OK
cannot build derivation '/nix/store/d9fldbn282fj7yxq3anfphpkh93barqv-OctoPrint-1.3.6.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/d9fldbn282fj7yxq3anfphpkh93barqv-OctoPrint-1.3.6.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=02cb919e-7a9d-4bc7-b2bc-c9a3488b9b4b)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```
tests/components/test_init.py ................                           [ 72%]
tests/components/test_introduction.py .                                  [ 73%]
tests/components/test_logger.py ....                                     [ 75%]
tests/components/test_script.py .....                                    [ 78%]
tests/components/test_shell_command.py ......                            [ 82%]
tests/components/test_system_log.py ...............                      [ 91%]
tests/components/test_websocket_api.py ..............                    [100%]

========================= 163 passed in 28.64 seconds ==========================
[31;1merror:[0m build of '/nix/store/7bq6jngcvwhj4v61x7nks3s3j035khji-bonfire-unstable-2017-01-19.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=b088a962-20fe-40a7-8769-cc73fefed14c)

Attempted: beancount, bonfire, buku, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

<details><summary>Partial log (click to expand)</summary><p>

```
tests/components/test_init.py ................                           [ 72%]
tests/components/test_introduction.py .                                  [ 73%]
tests/components/test_logger.py ....                                     [ 75%]
tests/components/test_script.py .....                                    [ 78%]
tests/components/test_shell_command.py ......                            [ 82%]
tests/components/test_system_log.py ...............                      [ 91%]
tests/components/test_websocket_api.py ..............                    [100%]

========================= 163 passed in 28.55 seconds ==========================
[31;1merror:[0m build of '/nix/store/7bq6jngcvwhj4v61x7nks3s3j035khji-bonfire-unstable-2017-01-19.drv' failed
```
</p></details>

Commit:
python.pkgs.python-jose: 1.3.2 -> 2.0.2

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38483&attempt_id=19a13d51-730a-4994-aee1-223762ac7032)

Attempted: beancount, home-assistant, mitmproxy, mkdocs, octoprint, pubs, python

The following builds were skipped because they don't evaluate on x86_64-darwin: bonfire, buku

<details><summary>Partial log (click to expand)</summary><p>

```
tests/components/test_logger.py ....                                     [ 75%]
tests/components/test_script.py .....                                    [ 78%]
tests/components/test_shell_command.py ......                            [ 82%]
tests/components/test_system_log.py ...............                      [ 91%]
tests/components/test_websocket_api.py ..............                    [100%]

========================== 163 passed in 4.99 seconds ==========================
building of '/nix/store/xl7sf65zyyrwmn55jsamxkvi6f35mmfk-python2.7-pyserial-3.4.drv' timed out after 1800 seconds
cannot build derivation '/nix/store/d9fldbn282fj7yxq3anfphpkh93barqv-OctoPrint-1.3.6.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/d9fldbn282fj7yxq3anfphpkh93barqv-OctoPrint-1.3.6.drv' failed
```
</p></details>

Commit:
python.pkgs.cmarkgfm: init at 0.3.0

Commit:
python.pkgs.readme_renderer: correct dependencies

Commit:
python.pkgs.redNotebook: remove alias

Commit:
python.pkgs.thinc: loosen version constraints

Commit:
python.pkgs.spacy: loosen version constraints

Commit:
python.pkgs.spacy: clean up propagatedBuildInputs

Commit:
python.pkgs.mozterm: disable on python 3

Commit:
python.pkgs.progressbar: disable on python 3

dotlambda commented:
Major packages that are still broken include
- pytest-flake8 (https://github.com/tholo/pytest-flake8/issues/45)
- pytest-sugar (https://github.com/Frozenball/pytest-sugar/issues/134)
- pandas
- (not really major, but I'd personally prefer that not to be broken) vobject (broken due to dateutil bump, see https://github.com/eventable/vobject/pull/111) The applications using it (radicale, papis, khard, ...) could pin the dateutil version as a workaround.

Commit:
python3.pkgs.git-annex-adapter: replace eject by utillinux

Commit:
platformio: move out of pythonPackages

Commit:
python.pkgs.click_5: uninit

Was only used by platformio, which now pins the click version using packageOverrides.

Commit:
platformio: 3.5.1 -> 3.5.2

Commit:
python.pkgs.protobuf: use 3.5 by default

dotlambda commented:
@FRidh Do you think it's better to make 3.5 the default global protobuf version or is it okay to make it the default for `pythonPackages` only.

Commit:
python: cmarkgfm: 0.3.0 -> 0.4.0

FRidh committed:
python: astroid: 1.6.2 -> 1.6.3

FRidh committed:
python: djangorestframework: 3.8.0 -> 3.8.2

FRidh committed:
python: dulwich: 0.19.1 -> 0.19.2

FRidh committed:
python: flit: 0.13 -> 1.0

FRidh committed:
python: ipython: 6.3.0 -> 6.3.1

FRidh committed:
python: Jug: 1.6.5 -> 1.6.6

FRidh committed:
python: mozrunner: 6.14 -> 6.15

FRidh committed:
python: openpyxl: 2.5.1 -> 2.5.2

FRidh committed:
python: pathspec: 0.5.5 -> 0.5.6

FRidh committed:
python: phonenumbers: 8.9.2 -> 8.9.3

FRidh committed:
python: pylint: 1.8.3 -> 1.8.4

FRidh committed:
python: robotframework: 3.0.2 -> 3.0.3

FRidh committed:
python: tornado: 5.0.1 -> 5.0.2

FRidh committed:
python: tqdm: 4.20.0 -> 4.21.0

FRidh committed:
python: buildbot-pkg: 1.1.0 -> 1.1.1

FRidh committed:
python: diceware: 0.9.4 -> 0.9.5

FRidh committed:
python.pkgs.pip: move expression

FRidh committed:
python: pip: 9.0.1 -> 9.0.3

FRidh committed:
python.pkgs.setuptools: 38.4.1 -> 39.0.1

FRidh committed:
python.pkgs.bootstrapped-pip: pip 9.0.1 -> 9.0.3 , setuptools 38.4.1 -> 39.0.1

dotlambda commented:
As mentioned in https://github.com/NixOS/nixpkgs/pull/38483/commits/d64839183c2a280d348598eb01960e697b080918, cairosvg should not be updated.

FRidh commented:
Oops.

I don't really know what to do with protobuf. We can *try* to upgrade the default Python version, but let's see if we can do that outside of this PR.

dotlambda commented:
I already updated it in https://github.com/NixOS/nixpkgs/pull/38483/commits/4d34eb71b5163c405c4fc5965b86265dff059f97 (included in this PR). It didn't introduce any build failure.

FRidh commented:
Pandas compatibility with dateutils 2.7 seems still quite far away. I don't think we should break pandas so let's revert the dateutils change.

Commit:
python.pkgs.cairosvg: 2.1.2 -> 2.1.3

purefn commented:
The update from aniso8601 2.0.1 to 3.0.0 broke the tests for flask-restful. Unfortunately there isn't a newer release of flask-restful that is compatible, but there is a [change on the master branch](https://github.com/flask-restful/flask-restful/commit/54979f0a49b2217babc53c5b65b5df10b6de8e05) which fixes the tests. Perhaps we should disable the tests for flask-restful for now?
